ceremonies/src/lib.rs

```diff
@@ -1669,6 +1731,79 @@ impl<T: Config> Pallet<T> {
 		(participant_votes, participant_attestations)
 	}
 
+	fn gather_meetup_data(
+		cid: &CommunityIdentifier,
+		participant: &T::AccountId,
+	) -> Result<
+		(CeremonyIndexType, MeetupIndexType, Vec<T::AccountId>, Location, T::Moment),
+		Error<T>,
+	> {
+		let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index();
+
+		let meetup_index = Self::get_meetup_index((*cid, cindex), participant)
+			.ok_or(Error::<T>::ParticipantIsNotRegistered)?;
+		let meetup_participants = Self::get_meetup_participants((*cid, cindex), meetup_index)
+			.ok_or(Error::<T>::GetMeetupParticipantsError)?;
+
+		let meetup_location = Self::get_meetup_location((*cid, cindex), meetup_index)
+			.ok_or(Error::<T>::MeetupLocationNotFound)?;
+
+		let meetup_time =
+			Self::get_meetup_time(meetup_location).ok_or(Error::<T>::MeetupTimeCalculationError)?;
+
+		Ok((cindex, meetup_index, meetup_participants, meetup_location, meetup_time))
+	}
+
+	fn add_attestations_to_registry(
+		participant: T::AccountId,
+		cid: &CommunityIdentifier,
+		cindex: CeremonyIndexType,
+		meetup_index: MeetupIndexType,
+		meetup_participants: &Vec<T::AccountId>,
```

Comment:
I have seen this for a while now. Instead of `&Vec<T::Account>` it is the better practice to put `&[T::Account]` because `&Vec<T::AccountId>` can always be coerced to `&[T::Account]`, but not vice versa.